tools/build-release-tarball

```diff
@@ -5,15 +5,27 @@ set -eu
 . "$(dirname "$0")"/lib/git-tools.bash
 
 usage() {
-    echo "Usage: $0 <ZULIP_VERSION>" >&2
+    cat <<EOF
+usage: $0 [--tag] <ZULIP_VERSION>
+
+Builds a release tarball from the current working tree, which must be
+clean.
+
+With --tag, also creates a signed git tag at the current commit.
+EOF
     exit 1
 }
 
-args="$(getopt -o '' -l 'help' -n "$0" -- "$@")"
-eval "set -- $args"
+args="$(getopt -o '' --long 'help,tag' -n "$0" -- "$@")"
+eval set -- "$args"
```

Comment:
It was clearer not to rely on `eval` taking multiple arguments, because all does with them is smash them together with spaces. It doesn’t do any quoting. Interpolating `$args` into this command is only safe because `getopt` is responsible for quoting the arguments appropriately.

Reply:
Thanks for catching that -- reverted.  That was a hunk leftover from when I was slowly diagnosing that BSD `getopt` is very much not GNU `getopt`, which was making running this on my local OSX machine very confusing.